pkg/fake/aksmachinesapi.go

```diff
@@ -535,7 +531,7 @@ func (c *AKSMachinesAPI) setDefaultMachineValues(machine *armcontainerservice.Ma
 	}
 
 	// Set ResourceID - simulates VM resource ID
-	// vmName = aks-<machinesPoolName>-<aksMachineName>-########-vm#
+	// vmName = aks-<machinesPoolName>-<aksMachineName>-########-vm
 	if machine.Properties.ResourceID == nil {
 		vmName := fmt.Sprintf("aks-%s-%s-12345678-vm0", agentPoolName, *machine.Name)
```

Comment:
This vm0 breaks unit tests for me; likely because GetAKSMachineNameFromVMName expects -vm suffix.

```suggestion
		vmName := fmt.Sprintf("aks-%s-%s-12345678-vm", agentPoolName, *machine.Name)
```

Reply:
> ProvisioningMode

It is now addressed in this PR. (that's why the default option change in a different PR was unusually smooth...)